.github/workflows/pipeline.yml

```diff
@@ -1,36 +1,28 @@
-name: lint and test
+name: Install and test
 
 on: [push, pull_request]
 
 jobs:
-  lint_and_test:
+  install_and_test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
 
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: ${{ matrix.python-version }}
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
 
       - name: Install poetry
-        run: pip install poetry==1.1.12
+        run: pip install poetry>=1.7.1
```

Comment:
I'd rather we pin to specific version in the CI pipeline please, just to avoid failures in the event of an incompatibility with poetry, for example if they release a new major version

Reply:
There is error with Python 3.12. Looking into details.

Reply:
Thanks, happy to pin to the latest version, think that is 1.8.2?

Reply:
I have pinned the poetry to latest version 1.8.2. Pipelines has passed.